compiler/resolution/functionResolution.cpp

```diff
@@ -6866,6 +6866,21 @@ static void resolveExternVarSymbols()
 }
 
 
+static void adjustInternalSymbols() {
+  SET_LINENO(rootModule);
+
+  // call _nilType nil so as to not confuse the user
+  dtNil->symbol->name = gNil->name;
+
+  // we want gRetByRefToken to be passable to 'ref' formals
+  makeRefType(dtRetByRefToken);
```

Comment:
It seems to me that it might be more reasonable to tell the compiler to not worry about coercing dtRetByRefToken / getting ref to it / etc.

Reply:
I disagree. That code does not need more special cases, IMO.